src/Application/Providers/AlarmsConditions/CreateAlarmSubscriptionProvider.cs

```diff
@@ -0,0 +1,273 @@
+        private AlarmFilterDefinition m_filter;
+
+        public CreateAlarmSubscriptionProvider(
+            IAlarmSubscriptionRepository subscriptionRepository,
+            ILogger<CreateAlarmSubscriptionProvider> logger,
+            IOptions<ConnectorConfiguration> connectorConfiguration,
+            AlarmMonitoredItemServiceInitializerFactoryDelegate opcMonitoredItemServiceInitializerFactory,
+            CreateAlarmSubscriptionsRequest command,
+            TelemetryMessageMetadata messageMetadata,
+            AlarmMonitoredItemValidator alarmMonitoredItemValidator) : base(command, connectorConfiguration, logger)
+        {
+            this._subscriptionRepository = subscriptionRepository;
+            this._opcMonitoredItemServiceInitializerFactory = opcMonitoredItemServiceInitializerFactory;
+            this._messageMetadata = messageMetadata;
+            this._alarmMonitoredItemValidator = alarmMonitoredItemValidator;
+            this._batchSize = this.Settings.OpcUa.AlarmSubscriptionBatchSize;
+
+            this._groupedItemsNotCreated = new Dictionary<string, List<string>>();
+        }
+
+        protected override async Task<string> ExecuteCommandAsync()
+        {
+            var errorMessages = await this.AddValidationErrorsAsync();
+            if (errorMessages.Any())
+            {
+                this.Logger.Debug($"Validation of {nameof(CreateAlarmSubscriptionsRequest.AlarmMonitoredItems)} was not successful.");
+                this.Logger.Trace(string.Join(" ", errorMessages));
+                return this.GetStatusMessage(errorMessages);
+            }
+
+            try
+            {
+                var subscriptionGroups = this.Command.AlarmMonitoredItems.GroupBy(item => item.PublishingInterval);
+
+                foreach (var group in subscriptionGroups)
+                {
+                    var groupItems = group.ToList();
+                    var batchHandler = new BatchHandler<AlarmSubscriptionMonitoredItem>(this._batchSize, this.SubscribeBatch());
+                    batchHandler.RunBatches(groupItems);
+                    this.Logger.Trace($"Alarm subscription with publishing interval {group.Key} ms: Subscribed to {groupItems.Count} nodes.");
+                }
+
+                foreach (var sub in this.Session.Subscriptions.Where(sub => !sub.PublishingEnabled))
+                {
+                    this.Logger.Trace($"Enabling publishing for alarm subscription {sub.Id}");
+                    sub.SetPublishingMode(true);
+                    sub.ConditionRefresh();
+                }
+            }
+            catch (Exception ex)
+            {
+                var error = ex.Demystify();
+                errorMessages.Add($"Bad: Could not create / update alarm subscriptions. {error.Message}");
+            }
+
+            //TODO: Implement subscription restore for alarms. Requires different Dto
+            //if (!base.Settings.DisableSubscriptionRestoreService && !errorMessages.Any())
```

Comment:
Place holder code for upcoming feature?

Reply:
Yes correct. Similar to the regular subscription restore, we will provide alarm subscription restore as well.
Will be done in a later story.